v-next/hardhat/src/internal/cli/init/subprocess.ts

```diff
@@ -0,0 +1,24 @@
+import type { LastParameter } from "../../../types/utils.js";
+
+import { spawn as nodeSpawn } from "node:child_process";
+
+export async function spawn(
+  command: string,
+  args: string[],
+  options: LastParameter<typeof nodeSpawn>,
+): Promise<void> {
+  console.log(command, args.join(" "));
+  const child = nodeSpawn(command, args, options);
+  await new Promise<void>((resolve, reject) => {
+    child.on("close", (code) => {
+      if (code !== 0) {
+        reject(
+          new Error(
+            `Command "${command} ${args.join(" ")}" exited with code ${code}`,
+          ),
+        );
```

Comment:
```suggestion
        );
        return;
```

if `code !== 0`, this promise will be both rejected and resolved otherwise.

Reply:
Great catch! Thank you 🙇